### ISSUE TITLE
Support ephemeral HDD storage

### ISSUE THREAD

**Issue by T4cC0re** [rancher/os#1612]
**RancherOS Version: v0.8.0-rc11**

**Where are you running RancherOS? baremetal**

Hi!

I would like to see the `rancher.state` extended in a way that it would format drives in `rancher.state.autoformat` on every boot.

The scenario I have is the following:
I have a PXE setup, where all nodes are to be 'stateless', however at least ephemeral state is needed to store containers and overlay data (`rancher.state`). So that upon reboot the disk is cleared again.

My proposal would be to check for a kernel parameter `rancher.state.ephermeral` and if that is truthy the initialization process will always format the disks in `rancher.state.autoformat`.

If you can give me a hint on where the code doing the checks is located I might be able to implement the feature myself and provide a pull-request.

Thanks in advance!

**Comment by SvenDowideit** [comment]
yes, I like that too - I want to use my RancherOS hosts as build slaves - so clean==reboot sounds useful.

**Comment by LLParse** [comment]
Just a thought - would a `tmpfs` filesystem be helpful? If your storage requirements aren't too hefty, that might be a viable option (and less work).

**Comment by T4cC0re** [comment]
(At least in my case) not really, as we need to be able to support big containers, as well as have temporary storage for builds.
We could use the HDD that is already in the chassis. And since auto format on empty disks is already supported this approach seems like a low hanging fruit to me.

**Comment by SvenDowideit** [comment]
yeah - me too - building RancherOS from scratch uses alot of disk space - and the memory in those boxes is needed for the build process, so an ephemeral disk would be pretty perfect.

as to where - `git grep autoformat` looks relevant to me :)

oh wow - that's still a shell script?

**Comment by T4cC0re** [comment]
Okay, so i started hacking on this, but I cannot get the shell-script to execute if the partition is already valid and I have no clue why...

Here you can find the patch to the changes I did (Hacky last-resorts included): http://pastebin.com/8qu8S59D

Can you lend me a hand?

**Comment by SvenDowideit** [comment]
so `dd if=/dev/zero of=${dev} bs=512 count=1` over-writes the disk's partitioning - if you do that, you'll need to re-partition the disk - which is why the mkfs.ext4 should then fail.

**Comment by T4cC0re** [comment]
The problem is not that the format does not work (it does work), but that the script is not executed on every boot.

**Comment by Shifter2600** [comment]
I too would like the sda always cleared on boot and everything rebuilt from the ipxe and cloud-config
